app/assets/src/components/Samples.jsx

```diff
@@ -35,7 +35,10 @@ class Samples extends React.Component {
     this.favoriteProjects = props.favorites || [];
     this.allProjects = props.projects || [];
     this.pageSize = props.pageSize || 30;
+    this.sampleAttributeHelper = {};
```

Comment:
Use more descriptive names, e.g. 
sampleAttributeHelper -> selectedSamples
sampleAttributeKeys -> sampleAttributesToStore

helps understand what these variables are for

Reply:
ok for sampleAttributeToStore. I feel selectedSamples has too much potential for confusion with selectedSampleIds.

Reply:
Plus, it isn't actually just selected samples. It could contain more than the selected samples

Reply:
You should merge both.
"It could contain more than the selected samples" How so?

Reply:
Ok now that's no longer the case. Still, I prefer to keep selectedSampleIds and sampleAttributeHelper separate. I need selectedSampleIds in the state to control visuals. sampleAttributeHelper is used for looking up certain additional information for the modal